src/main/scala/org/broadinstitute/dsde/workbench/sam/api/ResourceRoutes.scala

```diff
@@ -363,6 +364,17 @@ trait ResourceRoutes extends SamUserDirectives with SecurityDirectives with SamM
       }
     }
 
+  def putResourceAuthDomain(resource: FullyQualifiedResourceId, samUser: SamUser, samRequestContext: SamRequestContext): server.Route =
+    put {
+      requireAction(resource, SamResourceActions.readAuthDomain, samUser.id, samRequestContext) {
```

Comment:
This is one of the open questions - what permission level should we require? Do we need something new or can it be something like the resource owner can perform this?